src/index.js

```diff
@@ -1,14 +1,95 @@
 import './style.css';
 
-const li = document.querySelector('.last');
-const listData = [{ description: 'Wake up at 7:00am', completed: false, index: 0 },
-  { description: 'Work out and shower', completed: false, index: 1 },
-  { description: 'Take breakfast', completed: false, index: 2 },
-  { description: 'Meditation', completed: false, index: 3 },
-];
-function populateData() {
-  listData.forEach((data) => {
-    li.insertAdjacentHTML('beforebegin', `<li><p><input type="checkbox" id="checkbox">${data.description}</p><i class="fa-solid fa-ellipsis-vertical"></i></li>`);
+const addToDo = document.querySelector('.addToDo');
+const ul2 = document.querySelector('.ul-2');
+function getItems() {
+  const items = localStorage.getItem('To Do') ? JSON.parse(localStorage.getItem('To Do')) : [];
+  return items;
+}
+let toDoData = getItems(); // used let as the value would be changed.
+
+function setItems() {
+  const itemsJSON = JSON.stringify(toDoData);
+  localStorage.setItem('To Do', itemsJSON);
+}
+
+function displayList() {
+  ul2.innerHTML = '';
+  toDoData.forEach((data, i) => {
+    data.id = i;
+    const li = document.createElement('li');
+    li.className = 'added';
+    ul2.appendChild(li);
+    const p = document.createElement('p');
+    li.appendChild(p);
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.id = 'checkbox';
+    p.appendChild(checkbox);
+    const textContent = document.createElement('input');
+    textContent.className = 'input-text';
+    textContent.type = 'text';
+    textContent.value = data.description;
+    textContent.setAttribute('readonly', 'readonly');
+    p.appendChild(textContent);
+    const icon = document.createElement('i');
+    icon.className = 'fa-solid fa-ellipsis-vertical';
+    li.appendChild(icon);
   });
+  if (document.querySelectorAll('.added')) {
+    const added = document.querySelectorAll('.added');
+
+    added.forEach((el) => {
+      el.addEventListener('mouseover', () => {
+        el.classList.toggle('hover-list');
+        el.children[1].classList.toggle('fa-ellipsis-vertical');
+        el.children[1].classList.toggle('fa-trash-can');
+      });
+      el.addEventListener('mouseout', () => {
+        el.classList.toggle('hover-list');
+        el.children[1].classList.toggle('fa-trash-can');
+        el.children[1].classList.toggle('fa-ellipsis-vertical');
+      });
+    });
+  }
+}
+addToDo.addEventListener('change', () => {
+  if (addToDo.value) {
```

Comment:
* The [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=By%20default%20new,set%20to%201)..) states that 

> By default, new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
> Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
* This means that when you add todos to the list the index should be the same as the length of the array so that when you delete index item 5 which is the fifth item on the array the right item gets deleted, Kindly fix this issue